core/trino-main/src/main/java/io/trino/memory/ClusterMemoryManager.java

```diff
@@ -243,12 +244,14 @@ private synchronized void callOomKiller(Iterable<QueryExecution> runningQueries)
                 .map(this::createQueryMemoryInfo)
                 .collect(toImmutableList());
 
-        List<MemoryInfo> nodeMemoryInfos = nodes.values().stream()
-                .map(RemoteNodeMemory::getInfo)
-                .filter(Optional::isPresent)
-                .map(Optional::get)
-                .collect(toImmutableList());
+        Map<String, MemoryInfo> nodeMemoryInfosByNode = nodes.entrySet().stream()
+                .map(entry -> new NodeWithMemoryInfo(entry.getKey(), entry.getValue().getInfo()))
```

Comment:
nit: Since this is just a transient object maybe simply `Map.Entry<>`?

Reply:
Is there one with public consturctor?

Reply:
There's `AbstractMap#SimpleEntry`

Reply:
Thanks - actually I wanted to use Entry - but I did not find this one. Will refactor.